src/gh-webhook/github.ts

```diff
@@ -0,0 +1,111 @@
+import type { Env } from '..';
+
+export async function handleGitHubWebhook(request: Request, env: Env): Promise<Response> {
+  if (request.method !== 'POST') {
+    return new Response('Method not allowed', { status: 405, statusText: 'Method Not Allowed' });
+  }
+
+  const githubSecret = env.GITHUB_SECRET;
+  const webhookUrl = env.DISCORD_WEBHOOK;
+
+  if (!githubSecret || !webhookUrl) {
+    return new Response('Internal server error', { status: 500, statusText: 'Internal Server Error' });
+  }
+
+  const bodyText = await request.text().catch(() => null);
+  const headers = request.headers;
+
+  if (bodyText == null) {
+    return new Response('Failed to read request body', { status: 400, statusText: 'Bad Request' });
+  }
+
+  const authorized = await isAuthorized(headers, bodyText, githubSecret);
+
+  if (!authorized) {
+    return new Response('Unauthorized', { status: 401, statusText: 'Unauthorized' });
+  }
+
+  const json = JSON.parse(bodyText).catch(() => null);
```

Comment:
```suggestion
  let json: unknown | null;
  try {
    json = JSON.parse(bodyText);
  } catch {
    json = null;
  }
```
`.catch()` only works on [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)s